test/extended/kubevirt/util.go

```diff
@@ -353,23 +364,147 @@ func InKubeVirtClusterContext(oc *exutil.CLI, body func()) {
 	Expect(err).NotTo(HaveOccurred())
 
 	oc := exutil.NewHypershiftManagementCLI(hcpNamespace).AsAdmin()
 
-	mgmtClientSet := oc.KubeClient()
 	mgmtFramework.Namespace = &corev1.Namespace{
 		ObjectMeta: metav1.ObjectMeta{
 			Name: hcpNamespace,
 		},
 	}
-	mgmtFramework.ClientSet = mgmtClientSet
+	mgmtFramework.ClientSet = oc.KubeClient()
+	mgmtFramework.DynamicClient = oc.DynamicClient()
 
 	return oc
 }
 
+func virtualMachineInstanceClient(dc dynamic.Interface) dynamic.NamespaceableResourceInterface {
+	return dc.Resource(vmiGVR)
+}
+
+func virtualMachineInstanceMigrationClient(dc dynamic.Interface) dynamic.NamespaceableResourceInterface {
+	return dc.Resource(vmimGVR)
+}
+
+func composeVMIM(vmiNamespace, vmiName string) *unstructured.Unstructured {
+	vmim := unstructured.Unstructured{}
+	vmim.SetGroupVersionKind(vmimGVK)
+	vmim.SetNamespace(vmiNamespace)
+	vmim.SetGenerateName(vmiName + "-")
+	vmim.UnstructuredContent()["spec"] = map[string]interface{}{
+		"vmiName": vmiName,
+	}
+	return &vmim
+}
+
+func migrateWorkers(f *e2e.Framework) error {
+	By("migrating hosted cluster workers")
+	vmiClient := virtualMachineInstanceClient(f.DynamicClient)
+	vmiList, err := vmiClient.Namespace(f.Namespace.Name).List(context.Background(), metav1.ListOptions{})
+	if err != nil {
+		return fmt.Errorf("failed retrieving workers VMs at namespace %s: %v", f.Namespace.Name, err)
+	}
+
+	vmimClient := virtualMachineInstanceMigrationClient(f.DynamicClient)
+	vmims := []*unstructured.Unstructured{}
+	for _, vmi := range vmiList.Items {
+		vmim := composeVMIM(vmi.GetNamespace(), vmi.GetName())
+		if vmim, err = vmimClient.Namespace(f.Namespace.Name).Create(context.Background(), vmim, metav1.CreateOptions{}); err != nil {
+			return fmt.Errorf("failed creating vmim %s/%s: %v", vmi.GetNamespace(), vmi.GetName(), err)
+		}
+		vmims = append(vmims, vmim)
+	}
+	for _, vmim := range vmims {
+		phase := ""
+		err := wait.PollImmediate(5*time.Second, 5*time.Minute, func() (done bool, err error) {
+			vmim, err = vmimClient.Namespace(f.Namespace.Name).Get(context.Background(), vmim.GetName(), metav1.GetOptions{})
+			if err != nil {
+				return false, err
+			}
+			statusIface, ok := vmim.UnstructuredContent()["status"]
+			if !ok {
+				return false, nil
+			}
+			status, ok := statusIface.(map[string]interface{})
+			if !ok {
+				return false, fmt.Errorf("bad vmim.status: %+v", statusIface)
+			}
+			phaseIface, ok := status["phase"]
+			if !ok {
+				return false, nil
+			}
+			phase, ok = phaseIface.(string)
+			if !ok {
+				return false, fmt.Errorf("bad vmim.status.phase: %+v", phaseIface)
+			}
+			return phase == "Succeeded" || phase == "Failed", nil
+		})
+		if err != nil {
+			return err
+		}
+		if phase == "Failed" {
+			dumpKubevirtArtifacts(f)
+			vmiName, _, _ := unstructured.NestedString(vmim.UnstructuredContent(), "spec", "vmiName")
+			return fmt.Errorf("migration failed for vmi %s", vmiName)
+		}
```

Comment:
I think we want to dump the results for anything that is `phase != "Succeeded". So even if the poll times out, it would be useful to know what the state of migration was when it timed out. Anything that prevents us from having to try and reproduce the failure locally, or guess why the migration didn't completed, will help us debug why this test failed quicker